surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,152 @@
+
+// 4. ... but handled asynchronously when the setTimeout callback is popped from
+// the event loop
+setTimeout(() => rejected.catch(() => console.error("an error occured")), 100);
+```
+
+An unhandled rejection is a rejection which hasn't been handled yet. It might be handled in the future, like example 4, but it might also stay unhandled forever (like example 3).
+
+## Are you currently using Promises, async/await, a mix, or neither?
+
+Assume using as creating your own `new Promise`, `Promise.resolve`, and `async function`; or using libraries that produce Promises (that you use with `.then`/`.catch`, or with `.await`)
+
+(Multiple choice)
+
+  - [ ] Promises
+  - [ ] async/await
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## How are you handling rejections today?
+
+When consuming Promises, async functions or thenables, which of the options below do you use to handle rejections?
+
+(Multiple choice)
+
+  - [ ] `.catch()`
+  - [ ] `try / catch` wrapping an `await` operation
+  - [ ] Ignore the rejection
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## Do you know that Node.js has a global handled for unhandled rejections (`process.on('unhandledRejection')`)? If so, do you use it?
+ 
+  - [ ] I use `process.on('unhandledRejection')`
+  - [ ] I don't use `process.on('unhandledRejection')`
+  - [ ] I didn't know `process.on('unhandledRejection')` existed
+
+## Do you know that Node.js has a flag to change the behavior of unhandled rejections `--unhandled-rejections`? If so, do you use it?
+
+  - [ ] I use `--unhandled-rejections` set to `'strict'`
+  - [ ] I use `--unhandled-rejections` set to `'warn'`
+  - [ ] I use `--unhandled-rejections` set to `'none'`
+  - [ ] I don't use `--unhandled-rejections`
+  - [ ] I didn't know `--unhandled-rejections` existed
+
+## Are you using Promises in customer facing, production applications?
+
+  - [ ] Yes
+  - [ ] No
+  - [ ] I don't know
+
+## Which of the use cases described below do you spend most of your time on?
+
+  - [ ] Third-party libraries
+  - [ ] Command-line tools
+  - [ ] Web servers
+  - [ ] Computation-heavy processing
+  - [ ] Desktop applications
+  - [ ] Other (please elaborate)
+
+## When a rejected promise doesn't have a catch handler, what is the behavior of your application today? 
+
+For example:
+
+```js
+async function error() {
+  throw new Error();
+}
+error();
+```
+
+  - [ ] Node.js logs a warning alongside a deprecation notice, execution continues
+  - [ ] Logs a warning, continue running, no deprecation warning
+  - [ ] Logs a warning, continue running, no deprecation warning, exits with an error code when program finishes
+  - [ ] Exit as soon as possible
+  - [ ] Other (please elaborate)
+
+## What should be the default Node.js behavior for unhandled rejections?
+
+Consider the following modes:
+
+  - `strict`: throw and crash as soon as possible (on nextTick). Can't be overriden by unhandledRejection listener
```

Comment:
"This is similar to what Node.js does when you pass --abort-on-uncaught-exception" ?

Reply:
This is not absolutely correct. It does not crash but it raises an uncaught exception by default.

Maybe change it to:

```  - `strict`: raise an uncaught exception (on nextTick) similar to `throw new Error()` that is not caught. `unhandledRejection` listeners do not prevent raising the exception```